Python/iSense.py

```diff
@@ -0,0 +1,310 @@
+TEXT = iSenseAPI.TEXT
+LOCATION = iSenseAPI.LOCATION
+
+INVALID_CREDENTIALS_MESSAGE = "Credentials object is invalid. It must be initialized with either (username and password) OR (contrib_key and contrib_name)"
+
+class Project(object):
+
+    def __init__(self, project_id): 
+
+        dictionary = iSenseAPI.getProject(project_id, recur=True)
+
+        # __ makes the variables private. That way they can only be accessed within this class
+        self.__id = project_id
+        self.__name = dictionary['name']
+        self.__url = dictionary['url']
+        self.__path = dictionary['path']
+        self.__hidden = dictionary['hidden']
+        self.__featured = dictionary['featured']
+        self.__like_count = dictionary['likeCount']
+        self.__content = dictionary['content']
+        self.__time_ago_in_words = dictionary['timeAgoInWords']
+        self.__created_at = dictionary['createdAt']
+        self.__owner_name = dictionary['ownerName']
+        self.__owner_url = dictionary['ownerUrl']
+        self.__data_set_count = dictionary['dataSetCount']
+        self.__data_set_ids = dictionary['dataSetIDs']
+        self.__field_count = dictionary['fieldCount']
+
+        # array of data set objects
+        self.__data_sets = []
+        for data_set in dictionary['dataSets']:
+            data_set_object = DataSet(data_set)
+            self.__data_sets.append(data_set_object)
+
+        self.__fields = []
+        self.__field_ids = []
+        for field in dictionary['fields']:
+            field_object = Field(field['id'], field['name'], field['type'], 
+                                unit=field['unit'], restrictions=field['restrictions'])
+            self.__field_ids.append(field['id'])
+            self.__fields.append(field_object)
+
+    def refresh(self):
+        self.__init__(self.__id)
+
+    def getId(self):
+        return self.__id
+
+    def getDataSets(self):
+        return self.__data_sets
+
+    def getDataSetById(self, data_id):
+        for d in self.__data_sets:
+            if d.getName() == data_id:
+                return d 
+
+        # None of the data sets match
+        return None
+
+    def getDataSetByName(self, name):
+        for d in self.__data_sets:
+            if d.getName() == name:
+                return d
+
+        # None of the data sets match
+        return None
+
+    def getFields(self):
+        return self.__fields
+
+    def getFieldIds(self):
+        return self.__field_ids
+
+    # Other can be None, units if field is a number, or an array of text restrictions if the field is a text field
+    def addField(self, credentials, name, fieldType, other):
+        # Only the owner of the project can add fields
+        if credentials.isUser():
+            response = iSenseAPI.createField(credentials.getUsername(), credentials.getPassword(), self.__id, name, fieldType, other)
+            self.refresh()
+            return response
+
+        return INVALID_CREDENTIALS_MESSAGE
```

Comment:
It would make more sense to return a valid field object instead of returning a dictionary of random values.